src/shared/view/nunjucks/filters/form.js

```diff
@@ -54,26 +54,16 @@ const mapFormField = (field) => {
  * @param  {String|null} value - the date value.
  * @return {Array} - array of items, one for day, month and year
  */
-const getFormDateItems = (value) => {
-  const [year, month, day] = (value || '').toString().split(/[- T]/g);
-
-  return [
-    {
-      classes: 'govuk-input--width-2',
-      value: day,
-      name: 'day'
-    },
-    {
-      classes: 'govuk-input--width-2',
-      value: month,
-      name: 'month'
-    },
-    {
-      classes: 'govuk-input--width-4',
-      value: year,
-      name: 'year'
-    }
-  ];
+const getFormDateItems = (values, items = ['day', 'month', 'year']) => {
+  const formValues = (values || '').toString().split(/[- T]/g);
+  return items.map((item, index) => {
+    return {
+      classes: item === 'year' ? 'govuk-input--width-4' : 'govuk-input--width-2',
+      // items.length -1 because the array starts at 0 not 1
+      value: formValues[(items.length - 1) - index],
```

Comment:
I see what you are saying but this will cause further problems up the chain because the month will be provided first if the year is not present so the month will be mapped to the year and day will be mapped to the year. For the abstraction period we expect the day-month not the month-day.